### PR TITLE
add click tracking to best bets

### PR DESCRIPTION
**Pull request by srsuddath**
## Description
This PR updates the click tracking connection to allow tracking of best bets
and
Reconnects the suggestions fetch function to the new suggestions endpoint

## Testing done
manual testing

### CODE REVIEW

File: src/applications/search/containers/SearchApp.jsx [hunk 1]
```diff
@@ -170,14 +170,14 @@ class SearchApp extends React.Component {
...
-    const searchClickTrackingEndpoint = `/search_click_tracking`;
     const encodedQuery = encodeURIComponent(query);
     const apiRequestOptions = {
       method: 'POST',
     };
+    const moduleCode = bestBet ? 'BOOS' : 'I14Y';
 
     apiRequest(
-      `${searchClickTrackingEndpoint}?position=${searchResultPosition}&query=${encodedQuery}&url=${encodedUrl}&user_agent=${userAgent}`,
+      `/search_click_tracking?position=${searchResultPosition}&query=${encodedQuery}&url=${encodedUrl}&user_agent=${userAgent}&module_code=${moduleCode}`,
```
Comment (srsuddath) [inline src/applications/search/containers/SearchApp.jsx:180]: removing unnecessary abstraction

**Review by ncksllvn — APPROVED** [review]

File: src/applications/search/containers/SearchApp.jsx (hunk 1 above)
Comment (ncksllvn) [reply src/applications/search/containers/SearchApp.jsx:188]: cross-linking to related issue in Sentry, http://sentry.vfs.va.gov/organizations/vsp/issues/34827/?query=is%3Aunresolved

**Comment by U-DON** [comment]
@srsuddath: I think these changes have resulted in an uptick in the following Sentry alerts from the `SearchMenu` component:
- `Object doesn't support property or method 'sort'` ([Sentry](http://sentry.vfs.va.gov/organizations/vsp/issues/35577/))
- `o.sort is not a function` ([Sentry](http://sentry.vfs.va.gov/organizations/vsp/issues/35583/))

I noticed while testing the search bar in staging that I sometimes get this rate limit error back from the typeahead request:
```
{
	"error": {
		"code": "OVER_RATE_LIMIT",
		"message": "You have exceeded your rate limit. Try again later or contact us at https://api.gsa.gov:443/contact/ for assistance"
	}
}
```

I think this suggests that we should have some error handling case where we're not always sorting whatever gets returned in the response. Would you be able to look into this?

**Comment by srsuddath** [comment]
sure thing, I will dig in and investigate

**Comment by ncksllvn** [comment]
Wow @U-DON amazing detective work. I looked everywhere in our code base for where that could be happening.

@srsuddath specifically you may just want to wrap your [apiRequest](https://github.com/department-of-veterans-affairs/vets-website/blob/176175bc1c82acd064341e66983114dbb7a7ebb1/src/platform/site-wide/user-nav/components/SearchMenu.jsx#L101) in a try/catch and in the catch block send the error to Sentry. Typically we would rely on the API monitoring to capture these errors but since we are going direct to search.gov's API we bypass our own monitoring. 

We also may want to further throttle how often we send requests to search.gov (like, maybe wait a full second after the user is done typing to send the request) since it seems rate limiting is a factor. They might have that documented somewhere or we can reach out to them.

**Comment by srsuddath** [comment]
I will have to reach out to them about this, as they don't have any sort of rate limit mentioned in their implementation guide (it is all of 1 page of info)

Also, this API request is currently wrapped in a try-catch, however I think their API might be returning a 200 with an error in the body, which is causing this issue

Either way I am opening a ticket to get this fixed on the double

**Pull request merged**